sdl_android/src/main/java/com/smartdevicelink/api/SdlManager.java

```diff
@@ -75,6 +79,27 @@
     private PermissionManager permissionManager;
     */
 
+	CompletionListener subManagerListener = new CompletionListener() {
+		boolean allSucceeded = true;
+		int subManagerCount = 1; // Update per amount of sub managers implemented
+		@Override
+		public void onComplete(boolean success) {
```

Comment:
This needs a synchronized lock. Submanagers will complete asynchronously and therefore possibly try to alter the same variables at the same time.